### PR TITLE
Added methods to save plots

### LINKED ISSUE #82: Add save methods in plotting classes
🐞 **Problem**

The plots, as of now can't be saved in png/jpg format 
🎯 **Goal**

💡 **Possible solutions**

Add a `save` method to all the `*GeodesicPlotters`

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by sashank27**
Fixes #82 

Please review @shreyasbapat

### CODE REVIEW

File: src/einsteinpy/plotting/geodesics_static.py [hunk 1]
```diff
@@ -47,6 +47,12 @@ def plot(self, pos_vec, vel_vec, end_lambda=10, step_size=1e-3):
     def show(self):
         plt.show()
 
+    def _save_plot(self, name=None):
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: The method need not be private. 
Further, Just `save` is enough!

File: src/einsteinpy/plotting/geodesics_static.py [hunk 2]
```diff
@@ -113,3 +119,9 @@ def plot(
         self.ax.set_aspect(1)
 
         return lines
+
+    def _save_plot(self, name=None):
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: Same here!

**Comment by shreyasbapat** [comment]
My suggestion is that you mock the `plt.savefig` method and test the `save` method with that.

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py [hunk 3]
```diff
@@ -52,3 +52,16 @@ def test_plot_show_shows_plot(mock_show):
     cl.plot(r, v, el, ss)
     cl.show()
     mock_show.assert_called_with()
+
+
+@mock.patch("einsteinpy.plotting.geodesics_static.ScatterGeodesicPlotter.save")
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: I think you didn't understand wat to mock. I meant you can mock:

`einsteinpy.plotting.geodesics_static.plt.savefig`

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 4]
```diff
@@ -40,3 +40,16 @@ def test_plot_calls_plot_attractor(mock_plot_attractor):
     cl.plot(r, v, el, ss)
     mock_plot_attractor.assert_called_with()
     assert cl._attractor_present
+
+
+@mock.patch("einsteinpy.plotting.geodesics_static.StaticGeodesicPlotter.save")
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Same here!

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py (hunk 3 above)
Comment (sashank27) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Ohh! Sorry, my bad.

Comment (sashank27) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: @shreyasbapat Could you please help me exactly how to do the mocking? Also, where should I put the parameters for `savefig`?

Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Oh! Sure. You can take help from the documentation: https://docs.python.org/3/library/unittest.mock-examples.html#patch-decorators

**Comment by shreyasbapat** [comment]
I think we can ignore the codecov here!. Thanks for the valuable contribution @sashank27 !

**Comment by shreyasbapat** [comment]
You are welcome to join the developers chat on riot! 
https://riot.im/app/#/room/#einsteinpy:matrix.org

**Pull request merged**